lib/vault/api/app_id.rb

```diff
@@ -0,0 +1,31 @@
+require "json"
+
+require_relative "secret"
+require_relative "../client"
+
+module Vault
+  class Client
+    # A proxy to the {AppId} methods.
+    # @return [AppId]
+    def app_id
+      @app_id ||= AppId.new(self)
+    end
+  end
+
+  class AppId < Request
+    # Create an authentication token.
+    #
+    # @example
+    #   Vault.app_id.login('foo', 'bar') #=> #<Vault::Secret lease_id="">
+    #
+    # @param [String] app_id
+    # @param [String] user_id
+    #
+    # @return [Secret]
+    def login(app_id, user_id)
```

Comment:
I think, as part of this request, we want to save the resulting secret back onto the system in `$HOME/.vault-token` and then also set the Client's .token parameter to the returned value.